proposals/output_files.md

```diff
@@ -0,0 +1,46 @@
+# Proposal: Image ID, Digest, and Tar File Location Configuration
+
+Relevant issues: [#1561](https://github.com/GoogleContainerTools/jib/issues/1561), [#1921](https://github.com/GoogleContainerTools/jib/pull/1921)
+
+## Motivation
+
+Currently Jib hardcodes the location of the files it generates during the build, such as the image ID, digest,
+and tar file (for tar builds). For some users, it would be useful to allow configuring the location of these
+output files in a way that best fits their workflows.
+
+## Current Configuration
+
+Jib currently doesn't allow configuring these locations, and instead it uses hardcoded defaults:
+
+- Image tar -> `${buildDir}/jib-image.tar`
+- Image ID -> `${buildDir}/jib-image.id`
+- Image digest -> `${buildDir}/jib-image.digest`
+
+## Proposed Configuration
+
+The proposal is to allow users to configure their build with the following rules:
+1. Extensions to the filename (like id, digest, tar) will not be automatically appended
+1. Existing files at the specified locations will be overwritten
```

Comment:
These are a good point I haven't thought about. For example, we mark `@OutputFile` for the tar in Gradle, and I believe this establishes the usual Gradle input-output dependency relation so that if nothing is changed, Gradle won't rebuild the tar. So does this propose that we take out `@OutputFile` and always regenerate it?

And now I don't remember what was good about allowing path outside the build directory. Should we reconsider this? For example, you can only specify a relative path anchored at the build output directory. But then, people can do `../../`. But it is easy to guard against going out of the build directory.

Reply:
>So does this propose that we take out `@OutputFile` and always regenerate it?

That shouldn't be necessary, the only time we'd want to force an overwrite is if any of the inputs change. If we remove `@OutputFile` then I think there's a chance we overwrite the tar with an identical tar, which is useless.

>And now I don't remember what was good about allowing path outside the build directory. Should we reconsider this?

Unless there's some sort of explicit vulnerability here then I don't think there's a problem with allowing it. We're introducing this change is to make the builds more customizable, and limiting the paths they can choose seems like it could just add another possible roadblock.

Reply:
Oh, I think I mistakenly thought the statement was saying "... will always be overwritten". "... overwritten if necessary" could make this clear.

Additionally, I was thinking if `@OutputFile` is given, Gradle will delete the file when running `clean`, as the input-output relation is automatically established and maintained. Is it not the case if the file is outside the build directory? Can you confirm if Gradle `clean` simply deletes its usual build directory (`build/`)?

Reply:
I thought `@OutputFile` cleans if you specify that you want to clean that task specifically. For example `./gradlew cleanMyCustomTask`. But yeah worth confirming what it is.

Reply:
Looks like if I build a tar outside the build directory, `./gradlew clean` does not delete it even though the path is marked with `@OutputFile`.

Reply:
@loosebazooka looks like that works, `./gradlew cleanJibBuildTar` deletes the tar.

Reply:
Maybe we should mark the digest and id files as `@OutputFile` as well?